jenkins-plugin/src/test/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepNoOptionalsTest.java

```diff
@@ -0,0 +1,99 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+
+import org.jenkinsci.plugins.pipeline.maven.publishers.PipelineGraphPublisher;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.remoting.RoleChecker;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.InboundAgentRule;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.RealJenkinsRule;
+
+import hudson.FilePath;
+import hudson.model.Result;
+import hudson.model.Slave;
+import hudson.remoting.Callable;
+import hudson.tasks.Maven;
+import jenkins.model.Jenkins;
+import jenkins.mvn.DefaultGlobalSettingsProvider;
+import jenkins.mvn.DefaultSettingsProvider;
+import jenkins.mvn.GlobalMavenConfig;
+
+public class WithMavenStepNoOptionalsTest {
+
+    @Rule
+    public RealJenkinsRule jenkinsRule = new RealJenkinsRule()
+        // .withLogger("org.jenkinsci.plugins.pipeline.maven", Level.ALL)
+        .omitPlugins(
+            "analysis-core", "bootstrap5-api", "checks-api", "commons-lang3-api", "commons-text-api", "configuration-as-code",
+            "echarts-api", "font-awesome-api", "mysql-api", "postgresql-api", "maven-plugin", "flaky-test-handler",
+            "htmlpublisher", "jacoco", "jgiven", "junit", "junit-attachments", "matrix-project",
+            "maven-invoker-plugin", "pipeline-build-step", "plugin-util-api", "findbugs", "tasks");
+
+    @ClassRule
+    public static InboundAgentRule agentRule = new InboundAgentRule();
+
+    @Test
+    public void maven_build_jar_project_on_master_succeeds() throws Throwable {
+        jenkinsRule.then(j -> setup(j), j -> build(j));
+    }
+
+    private static void build(JenkinsRule r) throws Throwable {
+        String pipelineScript = "node('mock') {\n" +
+                "    git 'https://github.com/jglick/simple-maven-project-with-tests'\n" +
+                "    withMaven() {\n" +
+                "        sh 'mvn verify -Dmaven.test.failure.ignore=true'\n" +
+                "    }\n" +
+                "}";
+
+        WorkflowJob pipeline = r.createProject(WorkflowJob.class, "build-on-master");
+        pipeline.setDefinition(new CpsFlowDefinition(pipelineScript, true));
+        WorkflowRun build = r.assertBuildStatus(Result.SUCCESS, pipeline.scheduleBuild2(0));
+        r.assertLogContains("BUILD SUCCESS", build);
+        // System.out.println(JenkinsRule.getLog(build));
+    }
+
+    private static void setup(JenkinsRule r) throws Throwable {
+        Slave agent = agentRule.createAgent(r, "mock");
+        r.waitOnline(agent);
+
+        String mavenVersion = "3.6.3";
+        FilePath buildDirectory = agent.getRootPath();
+        FilePath mvnHome = buildDirectory.child("apache-maven-" + mavenVersion);
+        agent.getChannel().call(new Callable<Void, Exception>() {
```

Comment:
I think you meant to use `MasterToSlaveCallable`, though it is not really necessary to do this remotely since the agent and controller in this case will share a filesystem.